### ISSUE TITLE
Config as a class attribute of class bot

### ISSUE THREAD

**Issue by Susannova** [Susannova/Discord_Bot#106]
I think it would be better to have the configuration as a class attribute instead of a global singleton. Then, for an example, you could connect the bot to multiple servers that have different configurations. But the decorators can't have cogs attributes, so as an example you cant rewrite [this ](https://github.com/Susannova/Discord_Bot/blob/a8ed8a306061243570654a8d0c26470e9107cd46/cogs/cog_play_requests.py#L28) to
```python
@checks.is_in_channels(self.bot.config.channel_ids.play_request)
```
, see [stackoverflow](https://stackoverflow.com/questions/11731136/class-method-decorator-with-self-arguments).
The solution given there doesn't work also because, if I'm right, the checks function can only have one argument, see [discord.py](https://discordpy.readthedocs.io/en/latest/ext/commands/api.html#checks)

You could rewrite ```checks.is_in_channels``` (see [here](https://github.com/Susannova/Discord_Bot/blob/a8ed8a306061243570654a8d0c26470e9107cd46/core/checks.py#L16)) like this
```python
def is_in_channels(*channel_ids_atr):
    async def predicate(ctx: commands.context):
        channel_ids = [getattr(ctx.cog.config.channel_ids, channel_id_atr) for channel_id_atr in channel_ids_atr]
        channel_ids.append(CONFIG.channel_ids.bot)
        return utility.is_in_channels(ctx.message, channel_ids)
    return commands.check(predicate)
```
but then ``channel_ids_atr`` has to be a tuple of strings, which wouldn't be nice too.

**Comment by thyanin** [comment]
I used the global singleton because it would be easier to share it between modules because you dont have to reinstantiate it or worry that someone accidentally does. But I learned now that you can just create an instance of the normal class and then only import that so it should be fine if we use a normal class. But still I think the reason why its a Singleton is valid. Not sure if we have to change that.

If you would run the bot on multiple servers wouldnt you just have different instances of the code running instead of managing it in code? Does that even work?

For the problem:
The checks function from discord.py is not the checks function we use directly. We have a module checks.py. Here we return the checks function from discord.py but we call the is_in_channels function from our checks.py. We can definetly use more than one parameter on checks.py functions. 
But i dont think we need to, do we?

```python
def is_in_channels(channel_names):
    async def predicate(ctx):
        channel_names.append(consts.CHANNEL_BOT)
        return utility.is_in_channels(ctx.message, channel_names)
    return commands.check(predicate)
```

So this:
```python
@checks.is_in_channels(self.bot.config.channel_ids.play_request)
```

Should just work, it got a list of channels before and now it gets a list of channel ids i guess?

**Comment by Susannova** [comment]
Closed in 55e6c14af0a6037ff911c7709cb991a18c07602e
